app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
+      data = event[:full_data]
+      vm = data[:vm]
+      cluster = data[:cluster]
+
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(vm[:href])
+      cluster_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(cluster[:href])
+
+      vm_data = [ems_ref.include?('/templates/'), ems_ref, vm[:id], event[:message].split(/\W/)[1]]
+      vm_hash = ManageIQ::Providers::Redhat::InfraManager::RefreshParser.create_vm_hash(vm_data)
+
+      old_cluster = EmsCluster.find_by(:ems_ref => cluster_ref)
+      vm_hash[:ems_cluster_id] = old_cluster[:id]
+
+      ems = ExtManagementSystem.find_by_id(event[:ems_id])
+      new_vm = ems.vms_and_templates.build(vm_hash)
+      new_vm.save!
```

Comment:
@blomquisg @Fryguy Can you comment on this portion of the PR that is creating an inventory VM outside of the inventory refresh logic?

Normally I think events would be captured early enough that the inventory collection would not know about the VM yet, but there is really no guarantee.  If the event processing gets backed up this even could be processed after inventory has already collected the VM so at the very least it should check if the VM exists in the `vms` table before adding here.

My real concern is understanding the change that both inventory and this eventing logic could generate duplicate VM instances for the same VM?  Does inventory check for the VM instance right before adding it which would limit the window of time where it could create a duplicate?  Is any chance of duplication unacceptable here?

Also, if the VM did get duplicated can inventory recover from that?